files/external_node_v2.rb

```diff
@@ -163,6 +165,8 @@ def generate_fact_request(certname, filename)
     rescue => e
       raise "Could not generate facts for Foreman: #{e}"
     end
+  else
+    $stderr.puts 'Last facts are older than last upload time, rerun puppet agent on host to get newer facts'
```

Comment:
Right, you would also receive the message during the initial node info request, prior to pluginsync and the catalog request (containing the facts upload, the main ENC call). If it's logged by Puppet Server though, then it probably isn't a good idea to have this message.